conformance/tests/httproute-redirect-port.go

```diff
@@ -38,70 +39,172 @@ var HTTPRouteRedirectPort = suite.ConformanceTest{
+			t.Fatalf("unexpected error finding TLS secret: %v", err)
+		}
+
+		testCases := []struct {
+			expectedResponse http.ExpectedResponse
+			https            bool
+		}{
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Port: "8083",
+					},
+					Namespace: ns,
+				},
 			},
-			RedirectRequest: &roundtripper.RedirectRequest{
-				Host: "example.org",
-				Port: "8083",
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port-and-host",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Host: "example.org",
+						Port: "8083",
+					},
+					Namespace: ns,
+				},
 			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-status",
-				UnfollowRedirect: true,
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port-and-status",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 301,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Port: "8083",
+					},
+					Namespace: ns,
+				},
 			},
-			Response: http.Response{
-				StatusCode: 301,
+			{
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/port-and-host-and-status",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{
+						Port: "8083",
+						Host: "example.org",
+					},
+					Namespace: ns,
+				},
 			},
-			RedirectRequest: &roundtripper.RedirectRequest{
-				Port: "8083",
+			{
+				// When requestRedirect.Port is set to 80 and the interpreted scheme is
+				// http, port 80 should not be ADDED to the response Location header.
+				// (The Location header MAY still result in having a port if the request
+				// Host header contains one.)
+				expectedResponse: http.ExpectedResponse{
+					Request: http.Request{
+						Path:             "/http-listener-on-80-and-redirect-to-80",
+						UnfollowRedirect: true,
+					},
+					Response: http.Response{
+						StatusCode: 302,
+					},
+					RedirectRequest: &roundtripper.RedirectRequest{},
+					Namespace:       ns,
+				},
 			},
-			Namespace: ns,
-		}, {
-			Request: http.Request{
-				Path:             "/port-and-host-and-status",
-				UnfollowRedirect: true,
+			{
+				// When requestRedirect.Port is not set and the interpreted scheme is
+				// http, port 80 should not be ADDED to the response Location header.
+				// (The Location header MAY still result in having a port if the request
+				// Host header contains one.)
```

Comment:
In general we avoid adding conformance tests for "SHOULD" because we're using [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt) terminology here where should is a recommendation and not a requirement. In this case I think it's acceptable to have a conformance test here if it accepts the port either being specified or unspecified, but more specifically, it must not be a different value like 81 or something.

If I'm understanding this specific test case correctly, it's only checking the status code. I think we probably want to check for more than that here.

Reply:
Thanks. I've tried to follow this with the newer changes.